cpp/tests/strings/combine/join_list_elements_tests.cpp

```diff
@@ -150,14 +148,26 @@ TEST_F(StringsListsConcatenateTest, ScalarSeparator)
 
   // With null replacement
   {
-    auto const results = cudf::strings::concatenate_list_elements(
+    auto const results = cudf::strings::join_list_elements(
       string_lv, cudf::string_scalar("+++"), cudf::string_scalar("___"));
     std::vector<const char*> h_expected{
       "a+++___+++ccc", nullptr, "___+++efgh+++ijk", "zzz+++xxxxx"};
     auto const expected =
       STR_COL{h_expected.begin(), h_expected.end(), nulls_from_nullptr(h_expected)};
     CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected, print_all);
   }
+
+  // Turn off separator-on-nulls
+  {
+    auto const results = cudf::strings::join_list_elements(string_lv,
+                                                           cudf::string_scalar("+++"),
+                                                           cudf::string_scalar(""),
+                                                           cudf::strings::separator_on_nulls::NO);
+    std::vector<const char*> h_expected{"accc", nullptr, "efgh+++ijk", "zzz+++xxxxx"};
```

Comment:
```suggestion
    std::vector<const char*> h_expected{"a+++ccc", nullptr, "efgh+++ijk", "zzz+++xxxxx"};
```

Reply:
Humnn, we want not to apply separator on null, but we forgot the case when we have null in the middle. So if we have `['a', null, 'ccc']` and I'm not sure if the output should be `'accc'` or `'a+++ccc'`.

Reply:
I do think the separator would expected in this case. This is just a bug. I've fixed it locally but not pushed it yet.

Reply:
Great. That also matches with Spark behavior (https://github.com/NVIDIA/spark-rapids/issues/63#issuecomment-844217138).